composer/trainer/trainer.py

```diff
@@ -222,6 +223,25 @@ def __init__(
             else:
                 self.ddp_sync_strategy = DDPSyncStrategy(ddp_sync_strategy)
 
+        self.evaluators = []
+        if isinstance(eval_dataloader, DataLoader):
+            eval_dataloader = DataSpec(eval_dataloader)
+        if isinstance(eval_dataloader, DataSpec):
+            default_evaluator = Evaluator(label="eval_dataset", dataloader=eval_dataloader, metrics=None)
+            self.evaluators = [default_evaluator]
+        else:
+            self.evaluators = []
+            assert isinstance(eval_dataloader, list)
+            for evaluator in eval_dataloader:
+                assert isinstance(evaluator, Evaluator)
```

Comment:
I'm a bit confused on why the list needs to be rebuilt...if it's a list then isn't it going to be a list of evaluators?